stdlib/public/Concurrency/Task.cpp

```diff
@@ -243,6 +243,19 @@ AsyncTask::~AsyncTask() {
   Private.destroy();
 }
 
+void AsyncTask::setTaskId() {
+  static std::atomic<uint64_t> Id(1);
+
+  // We want the 32-bit Job::Id to be non-zero, so loop if we happen upon zero.
+  uint64_t Next;
+  do {
+    Next = Id.fetch_add(1, std::memory_order_relaxed);
+    Id = Next & 0xffffffff;
```

Comment:
Isn't this going to update the static `Id` instead of `Job::Id`? There is an unfortunate name collision between the static and the `Job` field.

Reply:
Ugh, yes, I will definitely rename the static!

Reply:
Updated with less dumb.